### ISSUE TITLE
sci-libs/cblas-reference-20161223-r100: Overwrites /usr/include/cblas.h

### ISSUE THREAD

**Issue by marbre** [gentoo-science/sci#730]
With upgrading cblas-reference, the symlink `/usr/include/cblas.h` is overwritten:

```
 * Messages for package sci-libs/cblas-reference-20151113-r100:

 * eselect alternatives update returned "250"

 * Messages for package sci-libs/cblas-reference-20161223-r100:

 * This package will overwrite one or more files that may belong to other
 * packages (see list below). You can use a command such as `portageq
 * owners / <filename>` to identify the installed package that owns a
 * file. If portageq reports that only one package owns a file then do
 * NOT file a bug report. A bug report is only useful if it identifies at
 * least two or more packages that are known to install the same file(s).
 * If a collision occurs and you can not explain where the file came from
 * then you should simply ignore the collision since there is not enough
 * information to determine if a real problem exists. Please do NOT file
 * a bug report at https://bugs.gentoo.org/ unless you report exactly
 * which two packages install the same file(s). See
 * https://wiki.gentoo.org/wiki/Knowledge_Base:Blockers for tips on how
 * to solve the problem. And once again, please do NOT file a bug report
 * unless you have completely understood the above message.
 * 
 * Detected file collision(s):
 * 
 *      /usr/include/cblas.h
 * 
 * Searching all installed packages for file collisions...
 * 
 * Press Ctrl-C to Stop
 * 
 * None of the installed packages claim the file(s).
 * 
 * Package 'sci-libs/cblas-reference-20161223-r100' merged despite file
 * collisions. If necessary, refer to your elog messages for the whole
 * content of the above message
```

I recently upgraded to `/usr/include/cblas.h`, have `app-admin/eselect-1.4.8-r100` installed on my system and use atlas-threads as cblas provider. It is actually no big deal to resolve the problem locally, but maybe we should take a deeper look.

**Comment by gagern** [comment]
Hit me too. I tried deleting the header file in question, but then `eselect` complains:

```
>>> Installing (1 of 1) sci-libs/cblas-reference-20161223-r100::science
!!! Error: Refusing to overwrite /usr/include/cblas.h: is not a symlink (use --force to override)
!!! Error: Could not set provider reference for alternative cblas: see previous errors
Call stack:
    * alternatives_do_set (alternatives-common.bash:247)
    * alternatives_do_update (alternatives-common.bash:424)
    * do_update (alternatives.eselect:57)
    * check_do (core.bash:24)
    * do_action (core.bash:105)
    * main (eselect:311)
exiting
 * eselect alternatives update returned "250"
!!! Error: Refusing to overwrite /usr/include/cblas.h: is not a symlink (use --force to override)
!!! Error: Could not set provider reference for alternative cblas: see previous errors
Call stack:
    * alternatives_do_set (alternatives-common.bash:247)
    * alternatives_do_update (alternatives-common.bash:424)
    * do_update (alternatives.eselect:57)
    * check_do (core.bash:24)
    * do_action (core.bash:105)
    * main (eselect:311)
exiting
```

**Comment by kiwifb** [comment]
It reminds me of some recent changes in handling of the python symlinks. You get those problem if you have "collision-protect" or similar in `FEATURES`. Apparently "collision-protect" let you responsible for checking for all potential collisions. You should switch to "protect-owned", which will only get in your way and prevent the merge if a package owns the symlink.
I cannot remember where I have seen that conversion on bugzilla or the forum, it was related to the last python-exec and possibly eselect upgrade.

**Comment by gagern** [comment]
With “protect-owned” you'd have gotten the behavior I got after manually removing the file. Seems that symlink “belongs” to eselect. I do have eselect-1.4.8-r100::science installed here, so there is nothing to upgrade here unless I want to go for a live ebuild.

**Comment by jlec** [comment]
Fixed in  00be0b2f37ec7a30e7d5d321d483b484a189f9b
